### DISCUSSION TITLE
RPC error when running litd

### DISCUSSION THREAD

**Discussion by Gusep** [lightninglabs/lightning-terminal#677] [Troubleshooting]
Hello,

I'm not able to run litd properly. I have LND node running alongside litd on the same machine, these are my respective configurations:

lnd.conf
`
rpclisten=localhost:10009
rpcmiddleware.enable=true
`

lit.conf
`
remote.lnd.rpcserver=127.0.0.1:10009
remote.lnd.macaroonpath=/home/bitcoin/.lnd/data/chain/bitcoin/mainnet/admin.macaroon
remote.lnd.tlscertpath=/home/bitcoin/.lnd/tls.cert
httpslisten=0.0.0.0:8443
`

And when running litd I still get these errors:
`
[ERR] LNDC: Could not receive from interceptor stream: rpc error: code = Unknown desc = RPC middleware not enabled in config
[ERR] STAT: could not start the lit sub-server: could not start Lit: could not start litd sub-servers: rpc error: code = Unknown desc = RPC middleware not enabled in config
`
As a result, when I access web interface and try Connect to terminal I get this error message
![image](https://github.com/lightninglabs/lightning-terminal/assets/486981/47e06f99-3a9c-43b6-8147-f2e4428d79ce)

I'm probably missing something but can't find what is it. Any help appreciated!
Thanks!

**Comment by levmi** [comment]
Hmm.. it seems like it's not picking up the config for `rpcmiddleware.enable=true`. Did you change that recently? If so, could you try a restart of LND so that it registers that change?

**Reply by Gusep** [reply]
Thanks for the reply. After LND service restart it magically started to work, even though I'm pretty sure I restarted it after the config modification. Anyway, works now, thank you sir!
